### PR TITLE
Step 5

### PR DESCRIPTION
**Pull request by Kwameoab**
- Finish Step 5
- Finish Step 6

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java [hunk 1]
```diff
@@ -24,35 +24,70 @@
...
+      // Check that the input is between 0 and 50
+      if (maxComment < 0 || maxComment > 50) {
+        System.err.println("User choice is out of range: " + maxCommentsString);
+        maxComment = -1;
       }
-      json += "}";
-      return json;
+      
+      int currentComment = 0;
+      for (Entity entity : results.asIterable()) {
```
Comment (ljfranklin) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java]: Run formatter, indentation is off

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java [hunk 2]
```diff
@@ -24,35 +24,70 @@
...
+        if(maxComment != -1){
```
Comment (ljfranklin) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java]: If `maxComment` does equal -1, return an error in the request with `response.sendError`

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -58,27 +58,46 @@ <h1>Kwame Ampadu-Boateng's Portfolio</h1>
...
+      <form action="/jsonData">
+        <p>How many comments do you want to see?</p>
+        <input
+          type="number"
+          name="maxCount"
+          min="0"
+          max="50"
+          value="10"
+          id="maxForm"
+          onchange="getJson(this.value)"
```
Comment (ljfranklin) [inline portfolio/src/main/webapp/index.html]: Does the `onchange` do anything here? Doesn't look like `getJson` takes a parameter.

File: walkthroughs/week-3-server/examples/todo-list/src/main/webapp/script.js [hunk 4]
```diff
@@ -47,6 +47,9 @@ function createTaskElement(task) {
 /** Tells the server to delete the task. */
 function deleteTask(task) {
   const params = new URLSearchParams();
+  console.log(params);
```
Comment (ljfranklin) [inline walkthroughs/week-3-server/examples/todo-list/src/main/webapp/script.js]: Log lines like this are helpful for debugging, but remove them before submitting your PR

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java [hunk 5]
```diff
@@ -24,35 +24,70 @@
...
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.List;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/jsonData")
-public class DataServlet2 extends HttpServlet {
-    private ArrayList<String> comments = new ArrayList();
+
+public class DataServlet2 extends HttpServlet {  
+  // This an array List that will be used in DeletedataSerlet.java to delete all comments
```
Comment (ljfranklin) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java]: Stale comment, delete

**Review by ljfranklin — CHANGES_REQUESTED** [review]
This PR is a little large, try to split into smaller PRs next time

File: portfolio/src/main/webapp/index.html (hunk 3 above)
Comment (Kwameoab) [reply portfolio/src/main/webapp/index.html]: I just tested it out. Onchange does do what it is suppose to do; however,  "this.value" does not impact the functionality so I will remove it.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java (hunk 1 above)
Comment (Kwameoab) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java]: It turns out prettier does not automatically run on java file but there is a way to have it work on it. I ran it on all the java files in my portfolio now.

Comment (ljfranklin) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java]: Great! Can you share that in the group chat?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java [hunk 6]
```diff
@@ -14,45 +14,87 @@
...
+    // Check that the input is between 0 and 50
+    if (maxComment < 0 || maxComment > 50) {
+      System.err.println("User choice is out of range: " + maxCommentsString);
+      maxComment = -1;
+    }
+    if (maxComment == -1) {
+      response.sendError(
+        HttpServletResponse.SC_BAD_REQUEST,
+        "Can not use input"
+      );
```
Comment (ljfranklin) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java:64]: I think you need a `return;` after sending the error

**Review by ljfranklin — APPROVED** [review]
LGTM, address my comment above before merging

**Pull request merged**